lambda_src/lambda_function.py

```diff
@@ -150,9 +152,14 @@ def sync_flow(event: Any, context: Any = None) -> Dict[Text, Any]:
         )
     else:
         data_dumps = dumps({'data': res_data}, default=str)
-        response = {'statusCode': 200, 'body': data_dumps}
+        response = {
+            'statusCode': 200,
+            'body': base64.b64encode(gzip.compress(data_dumps.encode())).decode(),
+            'isBase64Encoded': True,
+            'headers': {'Content-Encoding': 'gzip'}
+        }
 
-    if len(response) > 6_000_000:
+    if len(response['body']) > 6_000_000:
```

Comment:
this is a good catch but let's change one thing at a time in a PR and make a separate one for this change